### PR TITLE
mox: init at 0.0.5

### PR DESCRIPTION
**Pull request by dit7ya**
###### Description of changes

modern full-featured open source secure mail server for low-maintenance self-hosted email 

https://github.com/mjl-/mox

NOTE: This is not tested beyond basic commands.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by sikmir** [comment]
Could you please set actual version:
```sh
results/mox/bin/mox version
(devel)
```

**Comment by dit7ya** [comment]
@sikmir - I am not sure how to set the version during buildtime - at least couldn't find any buildflags and there is this...

https://github.com/mjl-/mox/blob/f6ed860ccb4911e41dda374238d834f680006612/moxvar/version.go#L8

**Comment by sikmir** [comment]
I'm not sure what is a proper way to set version, I thought `"-X github.com/mjl-/mox/moxvar.Version=${version}"` should work, but it doesn't. Maybe we need something like https://github.com/NixOS/nixpkgs/commit/d2e710716a75a4e48ab86333f2c1766b07421c86.

**Comment by theduke** [comment]
I'd love to see this merged, any chance this can be picked up?

Note: `0.0.4` was released in the meantime.

**Comment by dit7ya** [comment]
@theduke @sikmir Added a patch for setting the version during buildtime.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -5875,6 +5875,8 @@ with pkgs;
 
   moosefs = callPackage ../tools/filesystems/moosefs { };
 
+  mox = callPackage ../servers/mox { };
```
Comment (sikmir) [inline pkgs/top-level/all-packages.nix:5878]:
```suggestion
  mox = callPackage ../servers/mail/mox { };
```

**Review by sikmir — APPROVED** [review]

**Pull request merged**